pkg/services/ngalert/schedule/schedule.go

```diff
@@ -85,7 +86,12 @@ func (sch *schedule) definitionRoutine(grafanaCtx context.Context, key models.Al
 						sch.log.Error("failed saving alert instance", "title", alertDefinition.Title, "key", key, "attempt", attempt, "now", ctx.now, "instance", r.Instance, "state", r.State.String(), "error", err)
 					}
 				}
-				_ = stateTracker.ProcessEvalResults(key.DefinitionUID, results, condition)
+				transitionedStates := stateTracker.ProcessEvalResults(key.DefinitionUID, results, condition)
+				alerts := FromAlertStateToPostableAlerts(transitionedStates)
+				err = sch.SendAlerts(alerts)
+				if err != nil {
+					sch.log.Error("failed to put alerts in the notifier", "count", len(alerts), "err", err)
+				}
```

Comment:
I chose not to return the error here because I don't think we should bail out of the evaluation routine for all results if one of them has an issue. We need to make sure that we're doing something, and perhaps something more than just logging (record a metric, perhaps?) but I think we should continue evaluation even when we have an issue saving to the DB. 

Open to differing opinions, though!